src/vcpkg/commands.env.cpp

```diff
@@ -127,31 +127,31 @@ namespace vcpkg
             }
         }
 
-        auto env = get_modified_clean_environment(extra_env);
+        ProcessLaunchSettings settings;
+        auto& env = settings.environment.emplace(get_modified_clean_environment(extra_env));
         if (!build_env_cmd.empty())
         {
 #if defined(_WIN32)
             env = cmd_execute_and_capture_environment(build_env_cmd, env);
```

Comment:
I don't think this one is fixable, since the intent here is for the user to be able to use the terminal inside, that means it has to be in their terminal window.

Reply:
Ah, this caller does not, because `cmd_execute_and_capture_environment` itself is what passes `CreateNewConsole::Yes`.